### PR TITLE
added idle action for climate

### PR DESCRIPTION
**Pull request by danielkucera**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

```
[01:39:01][VV][api.service:210]: send_climate_state_response: ClimateStateResponse {
  key: 3913343285
  mode: CLIMATE_MODE_HEAT
  current_temperature: 24
  target_temperature: 23.5
  target_temperature_low: 0
  target_temperature_high: 0
  away: NO
  action: CLIMATE_ACTION_IDLE
}
```

### CODE REVIEW

File: esphome/components/climate/climate_mode.h [hunk 1]
```diff
@@ -25,6 +25,8 @@ enum ClimateAction : uint8_t {
   CLIMATE_ACTION_COOLING = 2,
   /// The climate device is actively heating (usually in heat or auto mode)
   CLIMATE_ACTION_HEATING = 3,
+  /// The climate device is idle (applicable for any mode)
```
Comment (OttoWinter) [inline esphome/components/climate/climate_mode.h]: We should make more clear the difference between IDLE and OFF. A second sentence in this documentation comment would help.

**Review by OttoWinter — COMMENTED** [review]
Update looks mostly good - just some more documentation would be good to avoid developers getting confused about the different actions.

This will also need an update in two places:
 - aioesphomeapi: https://github.com/esphome/aioesphomeapi
 - the Home Assistant ESPHome integration (add `idle` to the `map_enum`)

**Comment by danielkucera** [comment]
@OttoWinter is it more clear now? I can prepare patches for the api and hass afterwards.

**Comment by OttoWinter** [comment]
It's good now 👍

Also, I just realized you also need to update a couple more places in the esphome codebase as well:

 - `ClimateCall::set_mode` - example https://github.com/esphome/esphome/pull/845/files#diff-61e21d4296f352a45513a2d419419e5cR116-R117
 - `ClimateTraits` - add a supports_idle option and update supports_mode(),  https://github.com/esphome/esphome/pull/845/files#diff-541e2f2d8267d7329c76aa961fcb4665R17, https://github.com/esphome/esphome/pull/845/files#diff-17271c6878bd9fbfe237dd78c248256eR24
 - update mqtt climate (optional, but low hanging fruit) - https://github.com/esphome/esphome/pull/845/files#diff-6b9079b16765b3d6fff78888d6a5505dR35

**Comment by DAVe3283** [comment]
This will be very welcome when it is working, especially now that Home Assistant is showing heating/cooling active state on the temperature graph. Currently, a heat- or cool-only climate component in ESPHome reports that it is continually heating or cooling even though it is not.

**Comment by OttoWinter** [comment]
@DAVe3283 Please explain what you mean - this change won't affect anything in how HA displays things.

**Comment by danielkucera** [comment]
@OttoWinter it might. This is how it displays idle heating: Screenshot_20191123-195934_Firefox

**Comment by DAVe3283** [comment]
In recent HA builds, they have started showing the time a climate object is heating or cooling. This is the green vertical bars in this screenshot:
hass1

For a ESPHome climate component, it shows "cooling" all the time (purple):
hass2
Even though the chiller only ran for about an hour (the temperature dip).

**Review by OttoWinter — APPROVED** [review]

**Comment by danielkucera** [comment]
Oh, perfect, thanks for finishing this @OttoWinter.

**Pull request merged**